lib/models/pr.js

```diff
@@ -153,70 +155,77 @@ class PR {
     get payload() {
         return this._payload;
     }
-        
+
     set payload(payload) {
         return this._payload = payload;
     }
 
     get installation() {
         return this._installation;
     }
-    
+
     get number() {
         return this._number;
     }
-    
+
     get owner() {
         return this._owner;
     }
-    
+
     get repo() {
         return this._repo;
     }
-    
+
     get id() {
         return this._id;
     }
-    
+
     get body() {
         return this._body = this._body || (this.payload.body || "").replace(/\r\n/g, "\n").trim();
     }
-    
+
     get processor() {
         return this.config.type.toLowerCase();
     }
-    
+
     get aws_s3_bucket() {
         if (process.env.ALLOW_MULTIPLE_AWS_BUCKETS == "no") {
             return process.env.AWS_BUCKET_NAME;
         }
         return this.owner == "whatwg" ? process.env.WHATWG_AWS_BUCKET_NAME : process.env.AWS_BUCKET_NAME;
     }
-    
+
     get isMultipage() {
         return !!this.config.multipage;
     }
 
     requiresPreview() {
         return !(/<!--\s*no preview\s*-->/.test(this.body));
     }
-    
+
+    async requestIncludes() {
+        this.includes = await scanIncludes(Head.fromPR(this).github_repo_url, this.config.src_file, this.config.type);
+    }
+
     touchesSrcFile() {
         let relFiles = this.relevantSrcFiles;
-        return this.files.some(f => relFiles.indexOf(f.filename) > -1);
+        return this.files.some(f => relFiles.has(f.filename));
     }
-    
+
     get relevantSrcFiles() {
-        return [this.config.src_file];
+        if (this.includes) {
+            return new Set(this.includes).add(this.config.src_file);
+        }
+        return new Set([this.config.src_file]);
```

Comment:
more of a nit (ignore if you disagree) but it's usually better to just return arrays in a API like this. Sets are generally only useful inside algorithms to avoid duplicates and perform simple "set" thins - but as accessor properties, arrays are better IMO. Also, this will violate `.relevantSrcFiles !== .relevantSrcFiles`.

Reply:
Yeah, I'd be more confortable with this too, tbh.

Reply:
This violates the `===` property for arrays too, right? And does even before this PR?

The `Set` is useful in the single caller of this function, in `touchesSrcFile`: it switches an O(N^2) algorithm to O(N).

This could just as easily be a method as a getter; I just left the kind of property that was already here.

Reply:
> This violates the === property for arrays too, right? And does even before this PR?

True. Maybe just rename it `function getRelevantSrcFiles()` or `deriveRelevantSrcFiles()`? At least idiomatically, it gives it a reason to return a new thing every time. 

> The Set is useful in the single caller of this function, in touchesSrcFile: it switches an O(N^2) algorithm to O(N).

True, but given that we are only dealing with files in the maybe tens of files, I don't think it makes a huge difference. If we were processing a thousands of files, it may have a noticeable performance impact... or even the JIT might treat it as a set under the hood or do some other magic optimization.

Reply:
Done.